### PR TITLE
Make getVisibleClientRect behave as expected by default

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR changes `DomUtils.getVisibleClientRect` to require passing an extra truthy argument in order to access the (generally) unexpected behaviour of sometimes returning the rects of child elements. All locations in the code that *actually* wanted this behaviour have been updated to continue to use it.

### CODE REVIEW

**Comment by smblott-github** [comment]
Could you provide more info as to why you're proposing this?  Is it because of (incorrect) use of `DomUtils.getVisibleClientRect` in the scroller and elsewhere?

**Comment by mrmr1993** [comment]
> Is it because of (incorrect) use of `DomUtils.getVisibleClientRect` in the scroller and elsewhere?

I'm afraid so. Before the visual mode ones popped up, I'd been trying to rename it to something less inviting. Since the child-less version is useful too, this is probably an easier and better way to handle it.

File: lib/dom_utils.coffee [hunk 1]
```diff
@@ -80,9 +84,7 @@ DomUtils =
 
         # eliminate invisible elements (see test_harnesses/visibility_test.html)
         computedStyle = window.getComputedStyle(element, null)
-        if (computedStyle.getPropertyValue('visibility') != 'visible' ||
-            computedStyle.getPropertyValue('display') == 'none')
-          continue
+        continue if computedStyle.getPropertyValue('visibility') != 'visible'
```
Comment (smblott-github) [inline lib/dom_utils.coffee]: Re. your earlier explanation: ok.
On this line, though, why are we dropping the test on `display`/`none`?

Comment (mrmr1993) [inline lib/dom_utils.coffee]: `Element.getClientRects()` returns no rects when `display: none;` is set on the element, so that check will never pass.

**Pull request merged**